### DISCUSSION TITLE
Changing cell opacity on hover / forced to re-render entire chart

### DISCUSSION THREAD

**Discussion by Camel** [recharts/recharts#2884] [Q&A]
Hey all -- great library. Thanks for all the work on it. I am stuck on one issue -- I'm trying to "highlight" a Cell on a stacked bar chart when a user hovers over it by increasing opacity, while decreasing opacity on all other Cells. Right now I'm setting a "hover target" variable by using the "onmouseenter" and "onmouseleave" event handlers on the Bar object. The issue is, I'm forced to re-render the entire chart every time a user changes the Cell over which he/she is hovering to render these new opacities. Is there any way I can just change the underlying cell style and not re-render the entire chart? Effectively conditionally render the fillOpacity on a Cell based on a given hover state? Thanks in advance.

**Comment by jokiefer** [comment]
any solutions for this?

**Comment by ckifer** [comment]
Hi sorry for not getting back to this @Camel 

@Camel and @jokiefer 

https://codesandbox.io/p/devbox/simple-bar-chart-forked-jyvxmf

Just add `opacity` to `activeBar` and you get this for free.
`activeBar` opacity is your active state, the default opacity is your non-active state.

```tsx
      <BarChart
        data={data}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis dataKey="name" />
        <YAxis />
        <Tooltip cursor={false} />
        <Bar dataKey="uv" fill="#B3CDAD" activeBar={{ opacity: 0.8 }} />
        <Bar dataKey="pv" fill="#FF5F5E" activeBar={{ opacity: 0.8 }} />
      </BarChart>
```

<img width="212" alt="image" src="https://github.com/user-attachments/assets/e5e9aa35-7a70-412c-bb82-15eecbf1776e" />
